src/ctype/CType.ts

```diff
@@ -58,6 +58,7 @@ export default class CType implements ICType {
     this.schema = cTypeInput.schema
     this.owner = cTypeInput.owner
     this.hash = cTypeInput.hash
+    this.schema.$id = `kilt:ctype:${CTypeUtils.getHashForSchema(this.schema)}`
```

Comment:
I added this here. Since we are checking in the error handling. I felt it has already been checked.